application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/DefaultNativeConnectionProvider.java

```diff
@@ -98,31 +100,44 @@ private static LocalNativeConnectionProvider establishConnection(
             final LocalNativeConnectionProvider.Builder builder,
             final String host,
             final int port,
-            final long timeout)
+            final long timeout,
+            final RetryPolicy retryPolicy)
     {
-        long startTime = System.currentTimeMillis();
-        long endTime = startTime + timeout;
-        while (endTime > System.currentTimeMillis())
+        for (int count = 1; count <= retryPolicy.getMaxAttempts(); count++)
         {
             try
             {
                 return builder.build();
```

Comment:
There are two return builder.build() in this function. Could refactoring would be in place? I mean, if the first one fails (even after retrying), wouldn't the risk of the last one failing as well be somewhat heightened?